lab-gurpreet/lib/server.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const prom = require('./promise');
+const cowsay = require('cowsay');
+
+http.createServer((req, res) => {
+  let parsed = url.parse(req.url, true);
+  if (parsed.pathname === '/') {
+    res.writeHead(200, {
+      'Content-Type': 'text/plain'
+    });
+    res.write('Hello!\nNow try: localhost:3000/api/cowsay with a query string like ?text=mooo!\n');
+    res.end();
+  } else if (parsed.pathname === '/api/cowsay') {
+    res.writeHead(200, {
+      'Content-Type': 'text/plain'
+    });
+    if (req.method === 'GET') {
+      if (!parsed.query || parsed.query == null || !parsed.query.text) {
+        res.writeHead(400, {
+          'Content-Type': 'text/plain'
+        });
+        res.write(cowsay.say({
+          text: 'bad request\n try: localhost:3000/api/cowsay?text=howdy'
+        }) + '\n');
+        res.end();
+      } else {
+        res.write(cowsay.say(parsed.query) + '\n');
+        res.end();
+      }
+    } else if (req.method === 'POST') {
+      if (!parsed.query || parsed.query == null || !parsed.query.text) {
+        res.writeHead(200, {
```

Comment:
Not quite correct, when I send a POST request via httpie, your request hangs because the JSON is being sent over in the request body as opposed to being in query property. So, if I run this command: http POST localhost:3000/api/cowsay text=hello, I should see your cow saying hello back to me.